src/frontends/onnx/frontend/src/op/multinomial.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/multinomial.hpp"
+
+#include <cmath>
+#include <random>
+#include <vector>
+
+#include "openvino/core/shape.hpp"
+#include "openvino/op/multinomial.hpp"
+
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace onnx {
+namespace op {
+namespace set_1 {
+ov::OutputVector multinomial(const ov::frontend::onnx::Node& node) {
+    const auto input = node.get_ov_inputs().at(0);
+    int sample_size = node.get_attribute_value<int>("sample_size", 1);
+
+    auto input_shape = input.get_partial_shape();
+    int batch_size = input_shape[0].get_length();
+    int class_size = input_shape[1].get_length();
+
+    std::vector<std::vector<int>> samples(batch_size, std::vector<int>(sample_size, 0));
+
+    for (int i = 0; i < batch_size; ++i) {
```

Comment:
This block is strange for me, I don't think you can use any "data" here because it is translation process, not an inference.